### DISCUSSION TITLE
Error while retrieving firmware file / 404 Client Error: Not Found for url

### DISCUSSION THREAD

**Discussion by Mojtra** [emsesp/EMS-ESP32#1029] [General]
Hi, I recieve the following error while flashing my ESP32:
`Unexpected error: Error while retrieving firmware file 'https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32/bin/bootloader_dio_40m.bin': 404 Client Error: Not Found for url: https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32/bin/bootloader_dio_40m.bin`

It looks like the file ending has changed from "bootloader_dio_40m.bin" to "bootloader_dio_40m.elf"

**Comment by MichaelDvP** [comment]
What flasher do you use? 
The official: https://github.com/emsesp/EMS-ESP-Flasher/releases/tag/v1.0.0
get the bootloader from https://github.com/espressif/arduino-esp32/tree/2.0.5/tools/sdk/esp32/bin

**Comment by Mojtra** [comment]
Thank you. I was able to flash my ESP32 board with the flasher from you link. I've previously used the windows version of the flasher from here [https://github.com/MichaelDvP/EMS-ESP_Flasher/releases](url), which is linked in the following dokumentation [https://emsesp.github.io/docs/#/Uploading-firmware](url)

**Comment by MichaelDvP** [comment]
Oh, thanks for the info, i'll update the link and remove this version from my GH.

**Comment by lixi-2077** [comment]
Thank you very much, nice
